src/Collision.cc

```diff
@@ -200,6 +218,73 @@ sdf::SemanticPose Collision::SemanticPose() const
       this->dataPtr->poseRelativeToGraph);
 }
 
+/////////////////////////////////////////////////
+Errors Collision::CalculateInertial(gz::math::Inertiald &_inertial,
+                                    const ParserConfig &_config)
+{
+  Errors errors;
+
+  // Set the density value for the collision material
+  if (this->dataPtr->sdf->HasElement("density"))
+  {
+    this->dataPtr->density = this->dataPtr->sdf->Get<double>("density");
+  }
+  else
+  {
+    // If the density element is missing, let the user know that a default
+    // value would be used according to the policy
+    Error densityMissingErr(
+      ErrorCode::ELEMENT_MISSING,
+      "Collision is missing a <density> child element. "
+      "Using a default density value of 1000.0 kg/m^3. "
+    );
+    enforceConfigurablePolicyCondition(
+      _config.WarningsPolicy(), densityMissingErr, errors
+    );
+  }
+
+  auto geomInertial =
+    this->dataPtr->geom.CalculateInertial(this->dataPtr->density,
+                                          _config, errors);
+
+  if (!geomInertial)
+  {
+    errors.push_back({ErrorCode::LINK_INERTIA_INVALID,
+        "Inertia Calculated for collision: " +
+        this->dataPtr->name + " seems invalid."});
```

Comment:
```suggestion
        this->dataPtr->name + " is invalid."});
```

Reply:
Corrected in [943b886](https://github.com/gazebosim/sdformat/pull/1299/commits/943b8865b2e2b427c656ef2f69720e8d190b1b85)